sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -63,6 +66,10 @@ case class JdbcType(databaseTypeDefinition : String, jdbcNullType : Int)
  */
 @DeveloperApi
 abstract class JdbcDialect extends Serializable with Logging{
+
+  private lazy val zoneId = DateTimeUtils.getZoneId(SQLConf.get.sessionLocalTimeZone)
+  private lazy val timestampFormatter = TimestampFormatter.getFractionFormatter(zoneId)
```

Comment:
Could you move this inside `compileValue`? Since the dialect object is a singleton, we cannot use `lazy val` here, I think.